pkg/drive/utils.go

```diff
@@ -59,5 +74,67 @@ func getFSUUIDFromDrive(drive *directcsi.DirectCSIDrive) string {
 //            Else, return errNotMounted
 // ----------------------------------------------------------------------------
 func verifyHostStateForDrive(drive *directcsi.DirectCSIDrive) error {
+
+	devName, err := sys.GetDeviceName(uint32(drive.Status.MajorNumber), uint32(drive.Status.MinorNumber))
+	if err != nil {
+		return err
+	}
+	if filepath.Base(drive.Status.Path) != devName {
+		return fmt.Errorf("path mismatch. Expected %s got %s", filepath.Base(drive.Status.Path), devName)
+	}
+
+	if !utils.IsCondition(drive.Status.Conditions,
+		string(directcsi.DirectCSIDriveConditionReady),
+		metav1.ConditionFalse, string(directcsi.DirectCSIDriveReasonLost), "") {
+		return fmt.Errorf("drive is lost = %v", drive.Status.Conditions)
+	}
+
+	runUdevDataMap, err := sys.ReadRunUdevDataByMajorMinor(int(drive.Status.MajorNumber), int(drive.Status.MinorNumber))
+	if err != nil {
+		return err
+	}
+	runUdevData, err := sys.MapToUdevData(runUdevDataMap)
+	if err != nil {
+		return err
+	}
+	device := &sys.Device{
+		Name:         filepath.Base(drive.Status.Path),
+		Major:        int(drive.Status.MajorNumber),
+		Minor:        int(drive.Status.MinorNumber),
+		Virtual:      strings.Contains(drive.Status.Path, "/virtual/"),
+		Partition:    runUdevData.Partition,
+		WWID:         runUdevData.WWID,
+		Model:        runUdevData.Model,
+		UeventSerial: runUdevData.UeventSerial,
+		Vendor:       runUdevData.Vendor,
+		DMName:       runUdevData.DMName,
+		DMUUID:       runUdevData.DMUUID,
+		MDUUID:       runUdevData.MDUUID,
+		PTUUID:       runUdevData.PTUUID,
+		PTType:       runUdevData.PTType,
+		PartUUID:     runUdevData.PartUUID,
+		UeventFSUUID: runUdevData.UeventFSUUID,
+		FSType:       runUdevData.FSType,
+	}
+
+	if !uevent.ValidateDevice(device, []*directcsi.DirectCSIDrive{drive}) {
```

Comment:
we need to move ValidateDevice to sys pkg as discussed

Reply:
it was erroring because of circular dependency

Reply:
both `pkg drive` and `pkg uevent` already works with sys pkg. so, there won't be cyclic dependency

Reply:
yes but the signature of the function is `func ValidateDevice(device *Device, directCSIDrive *directcsi.DirectCSIDrive) bool {`  and `directcsi` - > `"github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta3"` 

This leads to cyclic dependency 
```
package github.com/minio/directpv/cmd/directpv
	imports github.com/minio/directpv/pkg/client
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta1
	imports github.com/minio/directpv/pkg/matcher
	imports github.com/minio/directpv/pkg/sys
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta3
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta2
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta1: import cycle not allowed
package github.com/minio/directpv/cmd/directpv
	imports github.com/minio/directpv/pkg/client
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta1
	imports github.com/minio/directpv/pkg/matcher
	imports github.com/minio/directpv/pkg/sys
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta3
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta2
	imports github.com/minio/directpv/pkg/matcher: import cycle not allowed
package github.com/minio/directpv/cmd/directpv
	imports github.com/minio/directpv/pkg/client
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta1
	imports github.com/minio/directpv/pkg/matcher
	imports github.com/minio/directpv/pkg/sys
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta3
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta2
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta1: import cycle not allowed
package github.com/minio/directpv/cmd/directpv
	imports github.com/minio/directpv/pkg/client
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta1
	imports github.com/minio/directpv/pkg/matcher
	imports github.com/minio/directpv/pkg/sys
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta3
	imports github.com/minio/directpv/pkg/apis/direct.csi.min.io/v1beta2
	imports github.com/minio/directpv/pkg/matcher: import cycle not allowed

```